src/clusterfuzz/_internal/bot/fuzzers/syzkaller/runner.py

```diff
@@ -180,16 +186,42 @@ def repro(self, repro_timeout, repro_args):
     """
     logs.log('Running Syzkaller testcase.')
     additional_args = copy.copy(repro_args)
-    result = self.run_and_wait(additional_args, timeout=repro_timeout)
-    result.return_code = self._crash_was_reproducible(result.output)
-
-    if result.return_code:
-      logs.log('Successfully reproduced crash.')
-    else:
-      logs.log('Failed to reproduce crash.')
-    logs.log('Syzkaller repro testcase stopped.')
-    return engine.ReproduceResult(result.command, result.return_code,
-                                  result.time_executed, result.output)
+
+    for retry_count in range(REPRO_RETRY_MAX):
+      result = self.run_and_wait(additional_args, timeout=repro_timeout)
+      log_location = re.search(REPRODUCE_LOG_LOCATION_PATTERN, result.output)
+
+      # syz-crush stopped before capturing crash output
+      if not log_location:
+        continue
+
+      # syz-crush did not reproduce any crash
+      if not self._crash_was_reproducible(result.output):
+        continue
+
+      _type, log_index, log_dir = log_location.groups()  # pylint: disable=invalid-name
+
+      try:
+        with open(f'{log_dir}/reproducer{log_index}', 'r') as f:
+          logs.log('Successfully reproduced crash.')
+          return engine.ReproduceResult(
+              command=result.command,
+              return_code=1,
+              time_executed=result.time_executed,
+              output=f.read(),
+          )
+      except FileNotFoundError as e:
+        logs.log('Reproducer log was not found. Rerunning syz-crush', e)
+        continue
+
+    logs.log(f'Failed to reproduce crash after {retry_count} attempts.')
+    result.return_code = 0
```

Comment:
nit: unnecessary line?

Reply:
I like having additional lines between groups of logical steps. wanted to show there is a logical break after `continue`

Reply:
I mean the `result.return_code = 0` line, since it's not used afterwards.

Reply:
oh right. I forgot to remove that. yes that is unnecessary.